etl_pipeline/transform.py

```diff
@@ -0,0 +1,210 @@
+                                'precipitation_probability': 'precipitation_prob',
+                                'rain': 'rainfall',
+                                'temperature_2m': 'temperature',
+                                'wind_direction_10m': 'wind_direction',
+                                'wind_gusts_10m': 'wind_gusts',
+                                'wind_speed_10m': 'wind_speed',
+                                'weather_code': 'weather_code_id'})
+    return data
+
+
+def change_data_types(data: pd.DataFrame) -> pd.DataFrame:
+    """Convert incorrect datatypes to match database."""
+    data['forecast_timestamp'] = data['forecast_timestamp'].apply(
+        pd.to_datetime)
+    data['visibility'] = data['visibility'].apply(int)
+    return data
+
+
+def create_warning_list(forecast: tuple) -> list[dict]:
+    """Create a list of weather warnings associated with a forecast."""
+    warning_list = []
+    for i, alert in enumerate(forecast):
+        if alert != 4:
+            warning_list.append(
+                {'alert_type_id': i + 1, 'severity_type_id': alert})
+    if not warning_list:
+        warning_list = None
+    return warning_list
+
+
+def calculate_heat_alerts(temperature: float) -> int:
+    """Find extreme heat alerts amongst temperature data."""
+    if 32 <= temperature:
+        return 1
+    if 27 <= temperature < 32:
+        return 2
+    if 22 <= temperature < 27:
+        return 3
+    return 4
+
+
+def calculate_wind_alerts(wind_gust: float, wind_speed: float) -> int:
+    """Find extreme wind alerts amongst wind data."""
+    if 130 <= wind_gust or 80 <= wind_speed:
+        return 1
+    if 110 <= wind_gust < 130 or 65 <= wind_speed < 80:
+        return 2
+    if 90 <= wind_gust < 110 or 50 <= wind_speed < 65:
+        return 3
+    return 4
+
+
+def calculate_ice_alerts(temperature: float) -> int:
+    """Find extreme ice alerts amongst temperature and precipitation data."""
+    if -10 >= temperature:
+        return 1
+    if -5 >= temperature > -10:
+        return 2
+    if -3 >= temperature > -5:
+        return 3
+    return 4
+
+
+def calculate_lightning_alerts(lightning: pd.Series) -> int:
+    """Find extreme lightning alerts amongst lightning potential data."""
+    if 2500 <= lightning:
+        return 1
+    if 1000 <= lightning < 2500:
+        return 2
+    if 300 <= lightning < 1000:
+        return 3
+    return 4
+
+
+def calculate_snowfall_alerts(snowfall: pd.Series) -> int:
+    """Find extreme lightning alerts amongst lightning potential data."""
+    if 2 <= snowfall:
+        return 1
+    if 0.5 <= snowfall < 2:
+        return 2
+    if 0.1 <= snowfall < 0.5:
+        return 3
+    return 4
+
+
+def calculate_visibility_alerts(visibility: pd.Series) -> int:
+    """Find low visibility alerts amongst visibility data."""
+    if 20 >= visibility:
+        return 1
+    if 50 >= visibility > 20:
+        return 2
+    if 150 >= visibility > 0.5:
+        return 3
```

Comment:
Looking good, small suggestion, I think it's a bit more readable to have the largest element on the right hand-side and the smallest element on the left hand side of the inequality so 150 >= visibility > 0.5   would become 0.5 < visibility <= 150.

Reply:
fixed